src/app/hooks/useRequest.js

```diff
@@ -1,33 +1,28 @@
-import { useState, useEffect } from 'react';
+import { useState } from 'react';
 
 import { errorApiHandler } from '../../utils/errorHandler';
 
-const useRequest = ({ request, payload }) => {
+const useRequest = ({ request }) => {
   const [isLoading, setIsLoading] = useState();
-  const [submitError, setSubmitError] = useState([]);
+  const [submitError, setSubmitError] = useState();
   const [response, setResponse] = useState();
 
   const sendRequest = dataPayload => {
+    if (isLoading) {
```

Comment:
This seems like something you should validate before calling sendRequest, not something I'd put here. So as to keep a single responsability for this function